lib/LaTeXML/Util/Pathname.pm

```diff
@@ -196,20 +199,24 @@ sub pathname_is_reloadable {
 # we _could_ make relative...
 sub pathname_relative {
   my ($pathname, $base) = @_;
+  return unless defined $pathname;
   $pathname = pathname_canonical($pathname);
   return ($base && pathname_is_absolute($pathname) && !pathname_is_url($pathname)
     ? File::Spec->abs2rel($pathname, pathname_canonical($base))
     : $pathname); }
 
 sub pathname_absolute {
   my ($pathname, $base) = @_;
+  return unless defined $pathname;
   $pathname = pathname_canonical($pathname);
   return (!pathname_is_absolute($pathname) && !pathname_is_url($pathname)
     ? File::Spec->rel2abs($pathname, ($base ? pathname_canonical($base) : pathname_cwd()))
     : $pathname); }
 
 sub pathname_to_url {
-  my $relative_pathname = pathname_relative($_[0]);
+  my ($pathname) = @_;
+  return unless defined $pathname;
```

Comment:
Nice patch!

This commit can be used as a reference example for why [the Option type](https://learning-rust.github.io/docs/e3.option_and_result.html#Basic-usages-of-Option) in Rust is such a pleasant abstraction that standardizes away a range of boilerplate guarding code. No such luck in Perl sadly.